src/main/java/com/iota/iri/TransactionValidator.java

```diff
@@ -167,51 +170,50 @@ public void addSolidTransaction(Hash hash) {
     private Runnable spawnSolidTransactionsPropagation() {
         return () -> {
             while(!shuttingDown.get()) {
-                Set<Hash> newSolidHashes = new HashSet<>();
-                useFirst.set(!useFirst.get());
-                synchronized (cascadeSync) {
-                    if (useFirst.get()) {
-                        newSolidHashes.addAll(newSolidTransactionsTwo);
-                    } else {
-                        newSolidHashes.addAll(newSolidTransactionsOne);
-                    }
-                }
-                Iterator<Hash> cascadeIterator = newSolidHashes.iterator();
-                Set<Hash> hashesToCascade = new HashSet<>();
-                while(cascadeIterator.hasNext() && !shuttingDown.get()) {
-                    try {
-                        Hash hash = cascadeIterator.next();
-                        TransactionViewModel transaction = TransactionViewModel.fromHash(tangle, hash);
-                        Set<Hash> approvers = transaction.getApprovers(tangle).getHashes();
-                        for(Hash h: approvers) {
-                            TransactionViewModel tx = TransactionViewModel.fromHash(tangle, h);
-                            if(quietQuickSetSolid(tx)) {
-                                tx.update(tangle, "solid");
-                            } else {
-                                if (transaction.isSolid()) {
-                                    addSolidTransaction(hash);
-                                }
-                            }
-                        }
-                    } catch (Exception e) {
-                        log.error("Some error", e);
-                        // TODO: Do something, maybe, or do nothing.
-                    }
-                }
-                synchronized (cascadeSync) {
-                    if (useFirst.get()) {
-                        newSolidTransactionsTwo.clear();
-                    } else {
-                        newSolidTransactionsOne.clear();
+                propagateSolidTransactions();
+            }
+        };
+    }
+
+    void propagateSolidTransactions() {
```

Comment:
method should have explicit visibility

Reply:
The reason it doesn't have explicit visibility is because it shouldn't be `public` yet unfortunately it is used in tests so unfortunately `private` won't work...
So default `package` visibility is the best option.

This a common thing we did in my last workplace. You have a better idea?

Reply:
Fair enough, ignore this then.